diff --git a/core/shared/platform/esp-idf/platform_internal.h b/core/shared/platform/esp-idf/platform_internal.h
--- a/core/shared/platform/esp-idf/platform_internal.h
+++ b/core/shared/platform/esp-idf/platform_internal.h
@@ -20,7 +20,10 @@
 #include <arpa/inet.h>
 #include <sys/socket.h>
 #include <sys/uio.h>
 #include <dirent.h>
+#include <sys/ioctl.h>
+#include <sys/poll.h>
+#include <time.h>
 
 #include "esp_pthread.h"
 #include "esp_timer.h"
@@ -147,7 +150,7 @@
 /* The below types are used in platform_api_extension.h,
    we just define them to make the compiler happy */
-typedef int os_poll_file_handle;
-typedef unsigned int os_nfds_t;
-typedef int os_timespec;
+typedef struct pollfd os_poll_file_handle;
+typedef nfds_t os_nfds_t;
+typedef struct timespec os_timespec;
 
 static inline os_file_handle
